File: src/main/java/pl/mroz/buddiesapi/infrastructure/container/RentalConfig.java
package pl.mroz.buddiesapi.infrastructure.container;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pl.mroz.buddiesapi.domain.rental.RentalRepository;
import pl.mroz.buddiesapi.domain.rental.RentalDomainService;
import pl.mroz.buddiesapi.domain.rental.RentalDomainServiceFactory;

@Configuration
public class RentalConfig {

    @Bean
    RentalDomainService rentalDomainService(RentalRepository repository) {
        return RentalDomainServiceFactory.rentalDomainService(repository);
    }
}

File: src/main/kotlin/buddiesapi/infrastructure/container/RentalConfig.kt
package buddiesapi.infrastructure.container

import buddiesapi.domain.rental.RentalDomainService
import buddiesapi.domain.rental.RentalDomainServiceFactory
import buddiesapi.domain.rental.RentalRepository
import org.springframework.context.annotation.Bean

class RentalConfig {
    @Bean
    fun rentalDomainService(repository: RentalRepository): RentalDomainService {
        return RentalDomainServiceFactory.rentalDomainService(repository)
    }
}
